L1Trigger/L1TTrackMatch/plugins/L1TkElectronTrackProducer.cc

```diff
@@ -0,0 +1,340 @@
+      edm::Ptr<L1TTTrackType> matchedL1TrackPtr(L1TTTrackHandle, itrack);
+
+      const math::XYZTLorentzVector P4 = egIter->p4();
+      float trkisol = isolation(L1TTTrackHandle, itrack);
+      if (RelativeIsolation && et_ele > 0.0) {  // relative isolation
+        trkisol = trkisol / et_ele;
+      }
+
+      TkElectron trkEm(P4, EGammaRef, matchedL1TrackPtr, trkisol);
+
+      trkEm.setTrackCurvature(matchedL1TrackPtr->rInv());  // should this have npars? 4? 5?
+
+      //std::cout<<matchedL1TrackPtr->rInv()<<"  "<<matchedL1TrackPtr->rInv(4)<<"   "<<matchedL1TrackPtr->rInv()<<std::endl;
+
+      if (IsoCut <= 0) {
+        // write the L1TkEm particle to the collection,
+        // irrespective of its relative isolation
+        result->push_back(trkEm);
+      } else {
+        // the object is written to the collection only
+        // if it passes the isolation cut
+        if (trkisol <= IsoCut)
+          result->push_back(trkEm);
+      }
+    }
+
+  }  // end loop over EGamma objects
+
+  iEvent.put(std::move(result), label);
+}
+
+// ------------ method called once each job just before starting event loop  ------------
+void L1TkElectronTrackProducer::beginJob() {}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void L1TkElectronTrackProducer::endJob() {}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+L1TkElectronTrackProducer::beginRun(edm::Run& iRun, edm::EventSetup const& iSetup)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+L1TkElectronTrackProducer::endRun(edm::Run&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+L1TkElectronTrackProducer::beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+L1TkElectronTrackProducer::endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void L1TkElectronTrackProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+// method to calculate isolation
+float L1TkElectronTrackProducer::isolation(const edm::Handle<L1TTTrackCollectionType>& trkHandle, int match_index) {
+  edm::Ptr<L1TTTrackType> matchedTrkPtr(trkHandle, match_index);
+  L1TTTrackCollectionType::const_iterator trackIter;
+
+  float sumPt = 0.0;
+  int itr = 0;
+  for (trackIter = trkHandle->begin(); trackIter != trkHandle->end(); ++trackIter) {
```

Comment:
range-based loop

Reply:
leave as is to avoid changing code below